client/lib/wp/sync-handler/test/index.js

```diff
@@ -33,57 +30,61 @@ describe( 'sync-handler', () => {
 		( { SyncHandler, hasPaginationChanged } = require( '../' ) );
 		wpcom = new SyncHandler( handlerMock );
 	} );
+
 	beforeEach( () => {
 		responseData = {};
 		setLocalData( {} );
 	} );
-	after( function() {
-		mockery.disable();
-		mockery.deregisterSubstitute( 'localforage' );
-	} );
+
 	it( 'should call callback with local response', () => {
 		const {
 			postListKey,
 			postListParams,
 			postListLocalRecord,
 			postListResponseBody
 		} = testData;
-		const callback = sinon.spy();
+		const callback = spy();
 		setLocalData( {
 			[ postListKey ]: postListLocalRecord
 		} );
 		wpcom( postListParams, callback );
 		expect( callback.calledWith( null, postListResponseBody ) );
 	} );
-	it( 'should call callback with request response', () => {
+	it( 'should call callback with request response', ( done ) => {
 		const {
 			postListKey,
 			postListParams,
 			postListResponseBody
 		} = testData;
-		const callback = sinon.spy();
+		const callback = spy();
 		responseData[ postListKey ] = postListResponseBody;
 		wpcom( postListParams, callback );
-		expect( callback ).to.have.been.calledOnce;
-		expect( callback.calledWith( null, postListResponseBody ) );
+		defer( () => {
```

Comment:
This and the next test were failing without defer. I guess it's because they both work with Promises. Though I have no idea how it was green before ;)